crates/ty_python_semantic/src/hash.rs

```diff
@@ -0,0 +1,306 @@
+use rustc_hash::FxBuildHasher;
+use std::borrow::Borrow;
+use std::hash::Hash;
+
+/// Always use this instead of [`rustc_hash::FxHashSet`].
+/// This struct intentionally does not implement `(Into)Iterator` because the iterator's output order will be unstable if the set depends on salsa's non-deterministic IDs or execution order.
+/// Only use `unstable_iter()`, etc. if you are sure the iterator is safe to use despite that.
+#[derive(Debug, Clone, get_size2::GetSize)]
+pub struct FxHashSet<V>(rustc_hash::FxHashSet<V>);
+
+impl<V: Eq + Hash> PartialEq for FxHashSet<V> {
+    fn eq(&self, other: &Self) -> bool {
+        self.0 == other.0
+    }
+}
+
+impl<V: Eq + Hash> Eq for FxHashSet<V> {}
+
+impl<V> Default for FxHashSet<V> {
+    fn default() -> Self {
+        Self(rustc_hash::FxHashSet::default())
+    }
+}
+
+#[allow(unsafe_code)]
+unsafe impl<V: Eq + Hash + salsa::Update> salsa::Update for FxHashSet<V> {
+    unsafe fn maybe_update(old_pointer: *mut Self, new_value: Self) -> bool {
+        unsafe { rustc_hash::FxHashSet::maybe_update(&raw mut (*old_pointer).0, new_value.0) }
+    }
+}
+
+impl<V: Eq + Hash> FromIterator<V> for FxHashSet<V> {
+    fn from_iter<T: IntoIterator<Item = V>>(iter: T) -> Self {
+        Self(rustc_hash::FxHashSet::from_iter(iter))
+    }
+}
+
+impl<V> std::ops::Deref for FxHashSet<V> {
+    type Target = rustc_hash::FxHashSet<V>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
```

Comment:
Why is it okay to `Deref` `FxHashSet` but not `FxHashMap`?

Reply:
I simply forgot to revert. There is no longer any need to implement `Deref` for both `FxHash{Set, Map}`.